aeson-gadt-th.cabal

```
@@ -20,12 +20,12 @@ flag build-readme
 
 library
   exposed-modules: Data.Aeson.GADT.TH
-  build-depends: base >= 4.8 && < 4.13
+  build-depends: base >= 4.8 && < 4.14
                , aeson >= 1.4 && < 1.5
                , containers >= 0.5 && < 0.7
-               , dependent-sum >= 0.6.1 && < 0.7
+               , dependent-sum >= 0.6.1 && < 0.8
                , transformers >= 0.5 && < 0.6
-               , template-haskell >= 2.11.0 && < 2.15
+               , template-haskell >= 2.11.0 && < 2.16
                , th-abstraction >= 0.3.1.0 && < 0.4
                , th-extras >= 0.0.0.4 && < 0.1
```

```suggestion
               , th-extras >= 0.0.0.4 && < 0.1
  if impl(ghc < 8.2)
    build-depends: dependent-sum < 0.6.2.2
```

The combination of GHC 8.0 and `dependent-sum 0.6.2.2` (and above) has never worked, because `dependent-sum` reexports `some` which does not reexport the `Some` pattern synonym on GHC 8.0, that `aeson-gadt-th` uses. (This seems to be a bug in `some`, where it does a `#if __GLASGOW_HASKELL__ >= 801` instead of `#if __GLASGOW_HASKELL__ >= 800`.)

The short-term fix is to forbid that combination, like the suggested patch does.

See this commit for successful build with the suggested change: https://github.com/Lysxia/aeson-gadt-th/commit/a2a2e17192191a2d81807ded54f22bbb9c36343b

A longer-term fix is to either patch `some` to export that constructor, or to patch `aeson-gadt-th` to not use it.